pkg/packagekit/wix/wix.go

```diff
@@ -0,0 +1,139 @@
+package wix
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Wix struct {
+	wixPath     string // Where is wix installed
+	packageRoot string // What's the root of the packaging files?
+	buildDir    string // The wix tools want to work in a build dir.
+	msArch      string // What's the microsoft archtecture name?
+
+	execCC func(context.Context, string, ...string) *exec.Cmd // Allows test overrides
+}
+
+func New(packageRoot, buildDir string) (*Wix, error) {
+	wix := &Wix{
+		wixPath:     `C:\wix311`, // FIXME: WTF go, why doesn't filepath.Join work? It produces relative paths
+		buildDir:    buildDir,
+		packageRoot: packageRoot,
+
+		execCC: exec.CommandContext,
+	}
+
+	switch runtime.GOARCH {
+	case "386":
+		wix.msArch = "x86"
+	case "amd64":
+		wix.msArch = "x64"
+	default:
+		return nil, errors.Errorf("unknown arch for windows %s", runtime.GOARCH)
+	}
+
+	return wix, nil
+}
+
+// InstallWXS takes an io.Reader and drops it's content into the builddir as installer.wxs
+func (wix *Wix) InstallWXS(installWXS []byte) error {
```

Comment:
this function doesn't add anything to the api, all it does is call WriteFile. Could just be a oneliner without the helper.